### PR TITLE
New Homepage

### LINKED ISSUE #7224: Add dapp shell pinning, search & quick-launch functionality
Builds on https://github.com/paritytech/parity/issues/7223

### PR DESCRIPTION
**Pull request by amaurymartiny**
Fix #7224:
- Only show dapps icon
- Add possibility to pin/unpin a dapp (quick-launch)
screen shot 2017-12-12 at 2 56 06 pm

- To browse all dapps, there a dapp called "Browse Dapps"
screen shot 2017-12-11 at 5 18 58 pm
- Possibility to add/remove dapp from homepage, or to open to try.

Feedback welcome.

[Edit] Updated scrot.

### CODE REVIEW

File: js/src/shellMiddleware.js [hunk 1]
```diff
@@ -18,75 +18,94 @@ import * as mobx from 'mobx';
...
         null,
         visibility
-          ? visibleStore.showApp(changeId)
-          : visibleStore.hideApp(changeId)
+          ? dappsStore.showApp(appId)
+          : dappsStore.hideApp(appId)
       );
       return true;
+    }
+    case 'shell_setAppPinned': {
```
Comment (jacogr) [inline js/src/shellMiddleware.js]: Empty line before to be consistent?

File: js/src/shellMiddleware.js [hunk 2]
```diff
@@ -18,75 +18,94 @@ import * as mobx from 'mobx';
...
-    case 'shell_setMethodPermissions':
+      callback(
+        null,
+        pinned
+          ? dappsStore.pinApp(appId)
+          : dappsStore.unpinApp(appId)
+      );
+      return true;
+    }
+    case 'shell_setMethodPermissions': {
```
Comment (jacogr) [inline js/src/shellMiddleware.js]: Empty line before to be consistent?

File: js/src/Dapps/dapps.js [hunk 3]
```diff
@@ -45,50 +46,82 @@ class Dapps extends Component {
...
+          ))
+        }
+      </div></div>
+  )
+
   render () {
-    const { availability } = this.props;
-    const applications = [].concat(this.store.visibleLocal, this.store.visibleViews, this.store.visibleBuiltin, this.store.visibleNetwork);
+    const pinned = this.store.pinnedApps; // Pinned apps
+    const unpinnedVisible = this.store.visibleApps.filter(app => // Visible apps that are not pinned
```
Comment (jacogr) [inline js/src/Dapps/dapps.js]: This should probably be in the store? (Can be computed?)

Comment (amaurymartiny) [reply js/src/Dapps/dapps.js]: Yes definitely.

**Comment by amaurymartiny** [comment]
Rebased & fixed comments.

**Comment by 5chdn** [comment]
are parity-news still a thing? :)

**Pull request merged**